driver/src/main/java/org/neo4j/driver/internal/net/BoltServerAddress.java

```diff
@@ -34,6 +34,9 @@
 {
     public static final int DEFAULT_PORT = 7687;
     public static final BoltServerAddress LOCAL_DEFAULT = new BoltServerAddress( "localhost", DEFAULT_PORT );
+    private static final Pattern LOCALHOST =
+            Pattern.compile( "^localhost$|^127(?:\\.[0-9]+){0,2}\\.[0-9]+$|^(?:0*\\:)*?:?0*1$", Pattern.CASE_INSENSITIVE );
```

Comment:
I though `127.0000.0.1` was supported but lets not think about that, I changed it back to a simpler pattern.